### PR TITLE
polish(zjow): impala cnn encoder refactor.

### PR DESCRIPTION
**Pull request by zjowowen**
1, vac encoder configuration polish.
2, encoder file polish.
3, enrich torch utils.
4, add relative unittest.

### CODE REVIEW

File: ding/model/template/vac.py [hunk 1]
```diff
@@ -63,40 +63,56 @@ def __init__(
...
+                    raise RuntimeError(
+                        "not support obs_shape for pre-defined encoder: {}, please customize your own encoder".
+                        format(obs_shape)
                     )
+
+        if self.share_encoder:
+            assert actor_head_hidden_size == critic_head_hidden_size, \
+                "actor and critic network head should have same size."
+            if encoder:
+                if type(encoder) is torch.nn.Module:
```
Comment (PaParaZz1) [inline ding/model/template/vac.py:97]: `if isinstance(encoder, torch.nn.Module):`

File: ding/model/template/vac.py [hunk 2]
```diff
@@ -63,40 +63,56 @@ def __init__(
...
-            assert share_encoder
-            self.share_encoder = share_encoder
-            self.encoder = encoder
+            if encoder:
+                if type(encoder) is torch.nn.Module:
+                    raise ValueError(
+                        "one encoder instance is not allowed \
+                        to be assigned to actor critic that not sharing encoders."
+                    )
+                else:
```
Comment (PaParaZz1) [inline ding/model/template/vac.py:110]: where is you encoder assignment branch, and add corresponding unittest

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: ding/model/template/vac.py (hunk 1 above)
Comment (zjowowen) [reply ding/model/template/vac.py]: fixed.

File: ding/model/template/vac.py (hunk 2 above)
Comment (zjowowen) [reply ding/model/template/vac.py:108]: fixed.
unittest for encoder assignment is added.

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**